src/iControlLBO/iControlLBO.psm1

```diff
@@ -49,20 +60,38 @@ function New-LBConnection
 		[Parameter(Mandatory=$true)]
 		[string]$username,
 
-		[string]$password
+		[string]$password,
+
+		[string]$key
 	)
 
-	if ($password -eq $null)
+	if ([string]::IsNullOrEmpty($password))
 	{
-		$serverCredential = Get-Credential -Username $username
+		if ($key -eq $null)
```

Comment:
@mynamewastaken @aaranmcguire Could we simplify this IF statement? It seems we're always asking for the password of the entered username, so we could just use one message? Or am I missing something?